### ISSUE TITLE
css are not picked up

### ISSUE THREAD

**Issue by atul-shimpi** [rails/rails#21099]
Rails - '4.2.3'
===============================
1] I have defined some global cascading styles in application.css and some controller specific styles in some other files in the same level as application.css. eg i have some styles in work.css. However, only styles in applicatin.css are picked up.
2] If i remove those css from application.css  and put them in sepeare file say global.css then both global.css and work.css are picked up.

**Comment by arunagw** [comment]
Do you have `require_self` in your application css file? Read more here http://guides.rubyonrails.org/asset_pipeline.html

**Comment by atul-shimpi** [comment]
require_self is inserted by rails generator when we create a new project.
No need to add in manually.
=========================================================
/*
 * This is a manifest file that'll be compiled into application.css, which will include all the files
 * listed below.
 *
 * Any CSS and SCSS file within this directory, lib/assets/stylesheets, vendor/assets/stylesheets,
 * or any plugin's vendor/assets/stylesheets directory can be referenced here using a relative path.
 *
 * You're free to add application-wide styles to this file and they'll appear at the bottom of the
 * compiled file so the styles you add here take precedence over styles defined in any styles
 * defined in the other CSS/SCSS files in this directory. It is generally better to create a new
 * file per style scope.
 *
 *= require_tree .
 *= require_self
 */

**Comment by meinac** [comment]
@atul-shimpi I couldn't reproduce the bug with `Rails 4.2.3`. Could you provide an example application which demonstrates the bug?

**Comment by atul-shimpi** [comment]
Env
----------------------------------------------
Rails 4.2.3
Ruby 2.2.0
Platform :  Ubuntu 12.04.4 LTS 64 bit
---------------------------------------------------
Steps
1] Create a new rails application 
2] Add controller named home with method index. Set this as root path in routes.rb
root 'home#index'.
Define a view for this action.
3] Add below css at top of applicaton.css (above require***)
body{color: red}
4] Create a new work.css file in app/assets/stylesheets.
Define below css in this file
#work{
    width: 100px;
    height: 100px;
    background-color: blue;
}
5] Add below line in home#index view
<div id="work">My Work</div>
6] Run the application. Observe that blue div is not displayed
7] Remove the body{color: red} from app/assets/stylesheets/work.css.
Refresh the browser. Observe the page. The blue div is diplayed.
8] Add body{color: red} below require in work.css and refresh the browser. The blue div is displayed.

In short, putting css above require is giving undesired behaviour.
They should be put only below require***.

**Comment by meinac** [comment]
Thanks for these steps, now I know how to create an example rails application!
> In short, putting css above require is giving undesired behaviour.
They should be put only below require***.

You have to give this information in description of the issue.
@rafaelfranca WDYT about this?

**Comment by rafaelfranca** [comment]
It is the know behavior. Css is a cascading language and the later
definitions take precedence over all before.

**Comment by rafaelfranca** [comment]
In fact the behavior is defined by the place where require_self is called, but the order order that the css is defined

**Comment by atul-shimpi** [comment]
The application.css has some very useful information. If we also inlcude this that will be useful. I had to spent almost an half day to figure this out.

**Comment by atul-shimpi** [comment]
Otherwise people will raise this issue again and again in the future.

**Comment by meinac** [comment]
@atul-shimpi If you think this is information will help others, you can always create a pull request.

**Comment by atul-shimpi** [comment]
Hey, thanks for pointing out that. Will do it.
